pilot/platform/cloudfoundry/servicediscovery.go

```diff
@@ -65,38 +68,30 @@ func (sd *ServiceDiscovery) GetService(hostname string) (*model.Service, error)
 	return nil, nil
 }
 
-func newService(hostname string) *model.Service {
-	return &model.Service{
-		Hostname: hostname,
-		Ports: []*model.Port{
-			{
-				Port:     AppPort,
-				Protocol: model.ProtocolHTTP,
-			},
-		},
-	}
-}
-
 // Instances implements a service catalog operation
 func (sd *ServiceDiscovery) Instances(hostname string, ports []string, tagsList model.LabelsCollection) ([]*model.ServiceInstance, error) {
 	resp, err := sd.Client.Routes(context.Background(), new(copilotapi.RoutesRequest))
 	if err != nil {
 		return nil, fmt.Errorf("getting instances: %s", err)
 	}
-	service := newService(hostname)
 	instances := make([]*model.ServiceInstance, 0, len(resp.GetBackends()))
 	backendSet, ok := resp.Backends[hostname]
 	if !ok {
 		return nil, nil
 	}
 	for _, backend := range backendSet.GetBackends() {
+		port := &model.Port{Port: sd.AppPort, Protocol: model.ProtocolHTTP}
```

Comment:
so now copilot is providing the app port? is that why you removed that code?

Reply:
yeah, but see https://github.com/istio/istio/pull/2714#discussion_r163684284